src/metatensor/models/cli/train_model.py

```diff
@@ -336,13 +341,43 @@ def _train_model_hydra(options: DictConfig) -> None:
         device_str=options["device"],
     )
 
-    save_model(model, options["output_path"])
+    save_model(model, f'{options["output_path"][:-3]}.ckpt')
+    export(model, options["output_path"])
+
+    logger.info("Setting up evaluation")
+    exported_model = torch.jit.load(options["output_path"])
+    requested_neighbors_lists = exported_model.requested_neighbors_lists()
+
+    # working around https://github.com/lab-cosmo/metatensor/issues/521
+    # Desired:
+    # for structure, _ in train_dataset:
+    #     attach_neighbor_lists(structure, requested_neighbors_lists)
+    # for structure, _ in validation_dataset:
+    #     attach_neighbor_lists(structure, requested_neighbors_lists)
+    # for structure, _ in test_dataset:
+    #     attach_neighbor_lists(structure, requested_neighbors_lists)
+    # Current:
+    train_dataloader = torch.utils.data.DataLoader(
+        train_dataset, batch_size=1, collate_fn=collate_fn
+    )
+    for (structure,), _ in train_dataloader:
+        attach_neighbor_lists(structure, requested_neighbors_lists)
+    validation_dataloader = torch.utils.data.DataLoader(
+        validation_dataset, batch_size=1, collate_fn=collate_fn
+    )
```

Comment:
Maybe the dataloader construction should be done in `_eval_targets` to avoid code duplication?

Reply:
We can't fix everything in this PR. Eval is hacky for now, but it was this way before. I'd like to open an issue for that

Reply:
But, these lines are new in this PR. I just wonder why they can't be in `_eval_targets` directly?